pkg/dependency/parser/java/pom/parse_test.go

```diff
@@ -1234,21 +1255,206 @@ func TestPom_Parse(t *testing.T) {
+					ID: "3ff14136-e09f-4df9-80ea-000000000002",
 					DependsOn: []string{
-						"org.example:example-dependency:1.2.3",
+						"3ff14136-e09f-4df9-80ea-000000000003",
 					},
 				},
 				{
-					ID: "org.example:example-dependency:1.2.3",
+					ID: "3ff14136-e09f-4df9-80ea-000000000003",
 					DependsOn: []string{
-						"org.example:example-api:2.0.0",
+						"3ff14136-e09f-4df9-80ea-000000000004",
+					},
+				},
+			},
+		},
+		{
+			name:      "multi module with deps with same GAV",
+			inputFile: filepath.Join("testdata", "multiple-modules-with-deps-with-same-gav", "pom.xml"),
+			local:     true,
+			want: []ftypes.Package{
+				{
+					ID:           "af5c1bca8a964540",
+					Name:         "com.example:root",
+					Version:      "1.0.0",
+					Relationship: ftypes.RelationshipRoot,
+				},
+				{
+					ID:           "9f9255f9043d446f",
+					Name:         "com.example:module1",
+					Version:      "1.0.0",
+					Relationship: ftypes.RelationshipWorkspace,
+				},
+				{
+					ID:           "84ab73e1b93aa7f5",
+					Name:         "com.example:module2",
+					Version:      "2.0.0",
+					Relationship: ftypes.RelationshipWorkspace,
+				},
+				{
+					ID:           "b80d1889cd7ec6f3",
+					Name:         "org.example:example-api",
+					Version:      "1.7.30",
+					Licenses:     []string{"The Apache Software License, Version 2.0"},
+					Relationship: ftypes.RelationshipDirect,
+				},
+				{
+					ID:           "b80d1889cd7ec6f3",
+					Name:         "org.example:example-api",
+					Version:      "1.7.30",
+					Licenses:     []string{"The Apache Software License, Version 2.0"},
+					Relationship: ftypes.RelationshipDirect,
+				},
+			},
+			//[INFO] --------------------------------[ pom ]---------------------------------
+			//[INFO]
+			//[INFO] --- dependency:3.7.0:tree (default-cli) @ module1 ---
+			//[INFO] com.example:module1:pom:1.0.0
+			//[INFO] \- org.example:example-api:jar:1.7.30:compile
+			//[INFO] --------------------------------[ pom ]---------------------------------
+			//[INFO]
+			//[INFO] --- dependency:3.7.0:tree (default-cli) @ module2 ---
+			//[INFO] com.example:module2:pom:2.0.0
+			//[INFO] \- org.example:example-api:jar:1.7.30:compile
+			//[INFO]
+			//[INFO] --------------------------------[ pom ]---------------------------------
+			//[INFO]
+			//[INFO] --- dependency:3.7.0:tree (default-cli) @ root ---
+			//[INFO] com.example:root:pom:1.0.0
+			//[INFO] ------------------------------------------------------------------------
+			wantDeps: []ftypes.Dependency{
+				{
+					ID: "af5c1bca8a964540",
+					DependsOn: []string{
+						"9f9255f9043d446f",
+						"84ab73e1b93aa7f5",
+					},
+				},
+				{
+					ID: "9f9255f9043d446f",
+					DependsOn: []string{
+						"b80d1889cd7ec6f3",
+					},
+				},
+				{
+					ID: "84ab73e1b93aa7f5",
+					DependsOn: []string{
+						"b80d1889cd7ec6f3",
+					},
+				},
+			},
+		},
+		{
+			name:      "multi module with similar deps, but different children",
```

Comment:
@knqyf263 I updated logic to use hash instead of UUID and found 1 issue.

I added this test case to show:

Based on our logic (https://github.com/aquasecurity/trivy/issues/7824#issuecomment-2446542674), we should use different IDs for all dependencies (because they can use different properties, depManagemets, etc.) - see `org.example:example-dependency` in this example.

But we use GAV in cache key only, which leads to artifact reuse and wrong packages.

I see 2 ways:
- abandon cache
- save pom.xml files before analyze to cache (pom.xml instead of `analysisResult`). In this case we will gain time for reading pom.xml files, but I'm not sure that it will help much.